test/models/watching_transaction_test.rb

```diff
@@ -0,0 +1,139 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class WatchingTransactionTest < ActiveSupport::TestCase
+  test 'create new watching transaction should work' do
+    params = attributes_for(:watching_transaction)
+
+    tx = WatchingTransaction.new(params)
+
+    user = create(:user)
+    tx.user_id = user.id
+
+    assert tx.save!,
+           'should save'
+
+    assert_not WatchingTransaction.new(params).save,
+               'should fail with the same data'
+  end
+
+  test 'watch should work' do
+    user = create(:user)
+    attrs = attributes_for(:watching_transaction)
+    ok, tx = WatchingTransaction.watch(user, attrs)
+
+    assert_equal :ok, ok,
+                 'should work'
+    assert_equal user.id, tx.user.id,
+                 'should have correct user id'
+  end
+
+  test 'resend should work' do
+    user = create(:user)
+    watching_transaction = create(:watching_transaction, user: user)
+    attrs = attributes_for(:watching_transaction)
+
+    ok, tx = WatchingTransaction.resend(user, watching_transaction, attrs)
+
+    assert_equal :ok, ok,
+                 'should work'
+    assert_equal watching_transaction.group_id, tx.group_id,
+                 'should have the same group as the previous transaction'
+  end
+
+  test 'resend transactions should keep all if none is mined' do
+    original = FactoryBot.create(:watching_transaction)
+    4.times do
```

Comment:
Instead of leaving at a fixed `4`, do `SecureRandom.rand(1..6)` to make the group sample independent of the size. Name this variable `size` or something similar to use below

Reply:
You can also create a random `group_id` and remove the `original` variable since the only thing being  used is its `group_id`.